server/src/main/java/com/decathlon/ara/cartography/SquashExporter.java

```diff
@@ -27,8 +54,117 @@ public String getFormat() {
+            log.debug("Full exception of the Squash export failure", ex);
+        }
         return new byte[0];
     }
+
+    @Override
+    public List<ExportField> listRequiredFields() {
+        List<ExportField> fields = new ArrayList<>();
+        fields.add(new ExportField(PROJECT_NAME, "Targeted project's name", "string",
+                "The name of targeted project in SquashTM. Note that the project must be created before doing the import."));
+        fields.add(new ExportField(USER, "Targeted User's name", "string",
+                "The name of the user whom will act as the creator of the requirements in SquashTM. Note that the user must exist before doing the import."));
+        return fields;
+    }
+
+    private Row createHeaderRow(Sheet sheet) {
+        Row header = sheet.createRow(0);
+        int rowIdx = 0;
+        this.addCellToRow(header, rowIdx++, "ACTION");
+        this.addCellToRow(header, rowIdx++, "PROJECT_ID");
+        this.addCellToRow(header, rowIdx++, "PROJECT_NAME");
+        this.addCellToRow(header, rowIdx++, "REQ_PATH");
+        this.addCellToRow(header, rowIdx++, "REQ_NUM");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_NUM");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_NAME");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CRITICALITY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CATEGORY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_STATUS");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_DESCRIPTION");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_#_TC");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_#_ATTACHEMENT");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CREATED_ON");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CREATED_BY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_LAST_MODIFIED_ON");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_LAST_MODIFIED_BY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_MILESTONE");
+        this.addCellToRow(header, rowIdx, "REQ_VERSION_CUF_<code du cuf>");
+        return header;
+    }
+
+    private void addRowToSheet(Sheet sheet, int index, String projectName, String path, String comment, String criticity, Date creationDate, String username) {
+        Row row = sheet.createRow(index);
+        this.addCellToRow(row, 0, "C");
+        this.addCellToRow(row, 2, projectName);
+        this.addCellToRow(row, 3, path);
+        this.addCellToRow(row, 4, "1");
+        this.addCellToRow(row, 5, "1");
+        this.addCellToRow(row, 7, criticity);
+        this.addCellToRow(row, 8, "CAT_UNDEFINED");
+        this.addCellToRow(row, 9, "WORK_IN_PROGRESS");
+        this.addCellToRow(row, 10, comment);
+        this.addCellToRow(row, 13, creationDate);
+        this.addCellToRow(row, 14, username);
+        this.addCellToRow(row, 15, creationDate);
+        this.addCellToRow(row, 16, username);
```

Comment:
same comment as above, except maybe replacing the array by a map ?

Reply:
For the moment a map will be less lisible and maintainable. If the format of the file change and more headers are added by Squash, why not 😄